napari/utils/key_bindings/legacy.py

```diff
@@ -0,0 +1,273 @@
+    Valid modifiers are Control, Alt, Shift, and Meta.
+
+    Letters will always be read as upper-case.
+    Due to the native implementation of the key system, Shift pressed in
+    certain key combinations may yield inconsistent or unexpected results.
+    Therefore, it is not recommended to use Shift with non-letter keys. On OSX,
+    Control is swapped with Meta such that pressing Command reads as Control.
+
+    Special keys include Shift, Control, Alt, Meta, Up, Down, Left, Right,
+    PageUp, PageDown, Insert, Delete, Home, End, Escape, Backspace, F1,
+    F2, F3, F4, F5, F6, F7, F8, F9, F10, F11, F12, Space, Enter, and Tab
+
+    Functions take in only one argument: the parent that the function
+    was bound to.
+
+    By default, all functions are assumed to work on key presses only,
+    but can be denoted to work on release too by separating the function
+    into two statements with the yield keyword::
+
+        @viewer.bind_key('h')
+        def hello_world(viewer):
+            # on key press
+            viewer.status = 'hello world!'
+
+            yield
+
+            # on key release
+            viewer.status = 'goodbye world :('
+
+    To create a keymap that will block others, ``bind_key(..., ...)```.
+    """
+    warnings.warn(
+        trans._(
+            'This function is deprecated and will be removed in version 0.6.0'
+        ),
+        DeprecationWarning,
+        stacklevel=2,
+    )
+
+    return func
+
+
+class KeymapProvider:
+    """Deprecated and will be removed in version 0.6.0.
+
+    Mix-in to add keymap functionality.
+
+    Attributes
+    ----------
+    class_keymap : dict
+        Class keymap.
+    keymap : dict
+        Instance keymap.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.keymap = {}  # type: ignore
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+
+        if 'class_keymap' not in cls.__dict__:
+            # if in __dict__, was defined in class and not inherited
+            cls.class_keymap = {}
+        else:
+            cls.class_keymap = {
+                coerce_keybinding(k): v for k, v in cls.class_keymap.items()
+            }
+
+    @classmethod
+    def bind_key(
+        cls,
+        key_bind: Union[KeyBindingLike, EllipsisType],
+        func=_UNDEFINED,
+        *,
+        overwrite=False,
+    ):
+        warnings.warn(
+            trans._(
+                'This function is deprecated and will be removed in version 0.6.0. Shortcuts set via the GUI will overwrite this.'
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
+        if key_bind is Ellipsis:
+            raise TypeError(
+                'Removed functionality: cannot use ellipsis as key binding.'
+            )
+
+        from napari._app_model._app import get_app
+
+        app = get_app()
+
+        kb = coerce_keybinding(key_bind)
```

Comment:
Just in case, checking locally the error from the CI, I think adding here a call to the `to_int` method could help. So something like:

```suggestion
        kb = coerce_keybinding(key_bind).to_int()
```